### PR TITLE
enable command line argument transformation

### PR DESCRIPTION
**Pull request by yaito3014**
this enables multioption like `poac build -rj8` via transforming `-rj10` into `-r -j 10` for known short options

### CODE REVIEW

**Review by ken-matsui — COMMENTED** [review]
Sorry for the late, thank you for working on this!  Just a few things need to be tweaked.

File: src/Cli.cc [hunk 1]
```diff
@@ -350,7 +350,53 @@ Cli::noSuchArg(std::string_view arg) const {
...
+            if (opt.shortName.empty())
+              continue;
+            if (opt.shortName.substr(1) != multioption.substr(i, 1))
+              continue;
+            if (opt.takeArgument) {
+              transformed.push_back(opt.shortName);
+              transformed.push_back(multioption.substr(i + 1));
+            } else {
+              transformed.push_back(opt.shortName);
+            }
```
[the remark this answers is not in the document]
Comment (yaito3014) [reply src/Cli.cc]: obviously redundant code. thanks.

File: src/Cli.hpp [hunk 2]
```diff
@@ -71,6 +71,7 @@ class Opt : public CliBase<Opt>, public ShortAndHidden<Opt> {
   std::string_view placeholder;
   std::string_view defaultVal;
   bool isGlobal = false;
+  bool takeArgument = false;
```
[the remark this answers is not in the document]
Comment (yaito3014) [reply src/Cli.hpp]: changed to use whether if placeholder is empty

File: src/Cmd/Common.hpp [hunk 3]
```diff
@@ -19,4 +19,5 @@ inline const Opt OPT_JOBS =
     Opt{ "--jobs" }
         .setShort("-j")
         .setDesc("Set the number of jobs to run in parallel")
-        .setDefault(NUM_DEFAULT_THREADS);
+        .setDefault(NUM_DEFAULT_THREADS)
+        .setTakeArgument(true);
```
[the remark this answers is not in the document]
Comment (yaito3014) [reply src/Cmd/Common.hpp]: added above `setDefault` as others do.

**Review by ken-matsui — APPROVED** [review]
This is a great feature!  Thank you!

**Pull request merged**